systemtests/src/main/java/io/enmasse/systemtest/AddressApiClient.java

```diff
@@ -51,35 +58,71 @@ public void createAddressSpace(AddressSpace addressSpace, String authServiceType
+        client.post(locEndpoint.getPort(), locEndpoint.getHost(), "/v1/addressspaces")
                 .timeout(20_000)
                 .as(BodyCodec.jsonObject())
                 .sendJsonObject(config, ar -> {
                     if (ar.succeeded()) {
                         responsePromise.complete(responseHandler(ar));
                     } else {
-                        Logging.log.error("Error creating address space {}", addressSpace, ar.cause());
+                        Logging.log.warn("Error creating address space {}", addressSpace);
                         responsePromise.completeExceptionally(ar.cause());
                     }
                 });
 
-        responsePromise.get(30, TimeUnit.SECONDS);
+        try {
+            responsePromise.get(30, TimeUnit.SECONDS);
+        } catch (Exception ex) {
+            if (ex.getCause() instanceof UnknownHostException && retry > 0) {
+                try {
+                    Logging.log.info("Trying to reload endpoint, remaining iterations: " + retry);
+                    createAddressSpace(addressSpace, authServiceType, addrSpaceType, openshift.getRestEndpoint(), retry - 1);
+                } catch (Exception ex2) {
+                    throw ex2;
+                }
+            } else {
+                ex.getCause().printStackTrace();
+                throw ex;
+            }
+        }
+    }
+
+    public void deleteAddressSpace(AddressSpace addressSpace) throws InterruptedException, TimeoutException, ExecutionException, UnknownHostException {
+        deleteAddressSpace(addressSpace, endpoint, initRetry);
     }
 
-    public void deleteAddressSpace(AddressSpace addressSpace) throws InterruptedException, TimeoutException, ExecutionException {
+    private void deleteAddressSpace(AddressSpace addressSpace, Endpoint locEndpoint, int retry) throws InterruptedException, TimeoutException, ExecutionException, UnknownHostException {
         CompletableFuture<JsonObject> responsePromise = new CompletableFuture<>();
-        client.delete(endpoint.getPort(), endpoint.getHost(), "/v1/addressspaces/" + addressSpace.getName())
+        client.delete(locEndpoint.getPort(), locEndpoint.getHost(), "/v1/addressspaces/" + addressSpace.getName())
                 .as(BodyCodec.jsonObject())
                 .timeout(20_000)
                 .send(ar -> {
                     if (ar.succeeded()) {
                         responsePromise.complete(responseHandler(ar));
                     } else {
-                        Logging.log.error("Error deleting address space {}", addressSpace, ar.cause());
+                        Logging.log.warn("Error deleting address space {}", addressSpace);
                         responsePromise.completeExceptionally(ar.cause());
                     }
                 });
-        responsePromise.get(30, TimeUnit.SECONDS);
+        try {
+            responsePromise.get(30, TimeUnit.SECONDS);
+        } catch (Exception ex) {
```

Comment:
Perhaps we could encapsulate this retry logic by creating a doHttp that takes a lambda wrapping the 'http operation' as the argument? The retry logic should be the same across all of these operations IMO.